compiler/rustc_arena/src/lib.rs

```diff
@@ -350,6 +351,20 @@ unsafe impl<#[may_dangle] T> Drop for TypedArena<T> {
 
 unsafe impl<T: Send> Send for TypedArena<T> {}
 
+#[inline(always)]
+fn align_down(val: usize, align: usize) -> usize {
+    assert!(align.is_power_of_two());
```

Comment:
One option to avoid this `assert` would be to have this take [`ptr::Alignment`](https://doc.rust-lang.org/nightly/std/ptr/struct.Alignment.html) instead.

([`Layout` has one of those internally](https://github.com/rust-lang/rust/blob/35636f9459720513ca4ed19373a4a7c9e0ea3c46/library/core/src/alloc/layout.rs#L49), though you might also need to add an accessor to get it as that type instead of just as `usize`.)

Reply:
These could probably be dropped to `debug_assert` also.